data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessor.java

```diff
@@ -0,0 +1,213 @@
+        for(String individualKey : individualMappings.keySet()){
+            if(NumberUtils.isParsable(individualKey)){
+                Float floatKey = Float.parseFloat(individualKey);
+                Range<Float> range = Range.between(floatKey, floatKey);
+                if(isRangeOverlapping(range)){
+                    String exceptionMsg = "map option contains key "+individualKey+" that overlaps with other range entries";
+                    throw new InvalidPluginConfigurationException(exceptionMsg);
+                }
+            }
+        }
+    }
+
+    private void parseFile(String filePath){
+        //todo
+    }
+
+    @Override
+    public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
+        //todo
+        for(final Record<Event> record : records) {
+            final Event recordEvent = record.getData();
+            if (Objects.nonNull(translateProcessorConfig.getMapWhen()) && !expressionEvaluator.evaluateConditional(translateProcessorConfig.getMapWhen(), recordEvent)) {
+                continue;
+            }
+            try {
+                String matchKey = record.getData().get(translateProcessorConfig.getSource(), String.class);
+                if(matchesIndividualEntry(record, matchKey) || matchesRangeEntry(record, matchKey) || matchesPatternEntry(record, matchKey)){
+                    continue;
+                }
+                else{
+
+                    // todo : add default, increment metrics, and/or add_tags
+
+                }
+            } catch (Exception ex){
+                LOG.error(EVENT, "Error mapping the source [{}] of entry [{}]",
+                        translateProcessorConfig.getSource(), record.getData(), ex);
+            }
+        }
+        return records;
+    }
+
+    public boolean matchesIndividualEntry(Record<Event> record, String matchKey){
+        if(individualMappings.containsKey(matchKey)){
+            record.getData().put(translateProcessorConfig.getTarget(), individualMappings.get(matchKey));
+            return true;
+        }
+        return false;
+    }
+
+    public boolean matchesRangeEntry(Record<Event> record, String matchKey){
+        if(!NumberUtils.isParsable(matchKey)){
+            return false;
+        }
+        Float floatKey = Float.parseFloat(matchKey);
+        for(Map.Entry<Range<Float>, String> rangeEntry : rangeMappings.entrySet()) {
+            Range<Float> range = rangeEntry.getKey();
+            if (range.contains(floatKey)) {
+                record.getData().put(translateProcessorConfig.getTarget(), rangeEntry.getValue());
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean matchesPatternEntry(Record<Event> record, String matchKey){
+        //todo
+        if(!Objects.nonNull(patternMappings)){
+            return false;
+        }
+        for(String pattern : patternMappings.keySet()){
+            if(Pattern.matches(pattern, matchKey)){
```

Comment:
This call will compile the patterns every single time, which will significantly degrade performance. We should compile the patterns on construction of the processor and use the compiled patterns here

Reply:
I will change the patternMatching logic in the next PR. I still need to implement some functionality in this method..